sebs/gcp/gcp.py

```
@@ -1429,174 +1610,519 @@ def download_execution_metrics(
+
+        Returns:
+            Storage source descriptor referencing the uploaded archive.
+        """
+        upload_info = self._generate_upload_url()
+        with open(package_path, "rb") as package_fp:
+            request = urllib.request.Request(
+                upload_info["uploadUrl"],
+                data=package_fp.read(),
+                method="PUT",
+                headers={"Content-Type": "application/zip"},
+            )
```

Ruff (0.15.12)

[error] Audit URL open for permitted schemes. Allowing use of `file:` or custom schemes is often unexpected.

(S310)